app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1268,6 +1270,27 @@ public void recenterMap(fr.free.nrw.commons.location.LatLng curLatLng) {
         mapBox.animateCamera(CameraUpdateFactory.newCameraPosition(position), 1000);
     }
 
+    @Override
+    public void showLocationOffDialog() {
+        // This creates a dialog box that prompts the user to enable location
+        DialogUtil
+            .showAlertDialog(getActivity(), getString(R.string.ask_to_turn_location_on), getString(R.string.nearby_needs_location),
+                getString(R.string.yes), getString(R.string.no),  this::openLocationSettings, null);
+    }
+
+    @Override
+    public void openLocationSettings() {
+        // This method opens the location settings of the device along with a followup toast.
+        Intent intent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+        try {
+            getContext().startActivity(intent);
+            Toast.makeText(getContext(), R.string.set_location_mode_high_accuracy, Toast.LENGTH_LONG).show();
+        } catch (Exception e) {
```

Comment:
We should never catch Exception, if you need to catch an exception catch the one that is actually thrown.

You probably don't need to catch an Exception here, you could [resolve the intent](https://stackoverflow.com/a/29994483)

Reply:
The try/catch clause was added because Android Studio was warning me that `startActivity may produce a Null Pointer Exception` but you are right. Let's resolve the intent.

Reply:
The NPE is because getContext is potentially null, a fragment only has a valid context after `onAttach` and it is removed from a context in `onDetach`